renderer/src/hooks/StationRewards.tsx

```diff
@@ -137,8 +145,34 @@ const useStationRewards = () => {
     }
   }, [])
 
+  const historicalRewardsWithLiveToday = useMemo(() => {
+    const rewards = [...historicalRewards]
+    let todayRewards = rewards[rewards.length - 1]
+    if (
+      !todayRewards ||
+      new Date(todayRewards.timestamp).getDate() !== new Date().getDate() ||
+      new Date(todayRewards.timestamp).getMonth() !== new Date().getMonth() ||
+      new Date(todayRewards.timestamp).getFullYear() !== new Date().getFullYear()
+    ) {
+      todayRewards = {
+        timestamp: new Date().toISOString(),
+        totalRewardsReceived: {
+          spark: rewards[rewards.length - 1]?.totalRewardsReceived.spark || 0,
```

Comment:
I am confused about this part, could you please explain at high level how is this fixing the problem?

Several lines above, I see that you call `https://stats.filspark.com/participant/${address}/scheduled-rewards` with `to=${yesterday}`.

That means we will never fetch rewards for today. The code here seems to apply yesterday's rewards for today. Shouldn't we fetch today's reward from somewhere?

Reply:
the goal is to show live _scheduled_ rewards of today, not total rewards received. Therefore, if there's no entry for today, I add a new entry and set its total rewards to the ones from the day before.

Then, below in line 170, we set the scheduled rewards of the most recent entry to the live value from the contract.

Reply:
I see. Would you mind capturing this information in a code comment?

```js
if (
      !todayRewards ||
      new Date(todayRewards.timestamp).getDate() !== new Date().getDate() ||
      new Date(todayRewards.timestamp).getMonth() !== new Date().getMonth() ||
      new Date(todayRewards.timestamp).getFullYear() !== new Date().getFullYear()
    ) {
      // If there's no entry for today, add a new entry copying data from yesterday.
      // We can use rewards from yesterday because we will change that value
      // later to the live value obtained from the smart contract
      todayRewards = {

Reply:
That's not what happening though: We're setting `totalRewardsReceived` from yesterday and will not update it afterwards. What we're doing is we're setting `totalScheduledRewards` to 0 and then afterwards update that with the live value from the contract.

We don't have total rewards as a live value in the contract.